### DISCUSSION TITLE
parallel nand flash interfacing with polarfire soc fpga using baremetal code

### DISCUSSION THREAD

**Discussion by Nileena00** [polarfire-soc/polarfire-soc-documentation#459] [Bare metal embedded software]
anybody know how to interface a parallel nand flash with polarfire soc and validate its communication using baremetal code

are there any drivers available?

**Comment by hughbreslin** [comment]
Hi @Nileena00 have you seen this [example](https://github.com/polarfire-soc/polarfire-soc-bare-metal-examples/tree/main/driver-examples/mss/mss-qspi/mpfs-qspi-w25n01gv-flash)?

**Reply by Nileena00** [reply]
> Hi @Nileena00 have you seen this [example](https://github.com/polarfire-soc/polarfire-soc-bare-metal-examples/tree/main/driver-examples/mss/mss-qspi/mpfs-qspi-w25n01gv-flash)?

but there the nand is using spi right?

**Reply by hughbreslin** [reply]
Ahh apologies, is this more what you're looking for: [CoreMemCtrl](https://www.microchip.com/en-us/products/fpgas-and-plds/ip-core-tools/corememctrl)? It should be available in the Libero catalog. There are no registers in the core, I haven't used it before but it looks like it transparently maps the flash with the AHB interface. So configuring the core correctly you should just be able to read and write to the flash using a FIC. Let me know if I've gotten mixed up - if you could share the flash you're looking at using it would be helpful :)

**Reply by Nileena00** [reply]
i am looking it for MT29F256G08AUCABH3

**Reply by hughbreslin** [reply]
Ahhh ok - I was going to point you towards the Microchip parts but we don't have 2Gbit density!

Personally I think your best bet is to go with the corememctrl IP and use the FPGA to handle the addressing and data. That way you don't need a driver and your flash just becomes any old memory with a base address you read and write to. Otherwise I'm not sure, you might have to use GPIOs or something to try and encode addresses and data which will be very slow. This is what I have in my head as the configuration:

![image](https://github.com/polarfire-soc/polarfire-soc-documentation/assets/44862881/a502628e-6aef-4c9b-a027-d82e06540b12)

This just don't have support for sending commands / reading parameters from the flash

**Reply by Nileena00** [reply]
ohh 

thankyou so much!
